### PR TITLE
rose macro and config-edit: namelist options fail if uppercase

### LINKED ISSUE #1569: rose macro: warn or fail if macro adds uppercase namelist options
If a macro adds an uppercase namelist option to a config, it is lowercased
silently on save as part of the normal config tidying. It would be good if the
user was made aware of the change via a warning, or if the uppercase was
spotted and the macro failed.

### PR DESCRIPTION
**Pull request by kaday**
Closes #1569 

@benfitzpatrick please review

### CODE REVIEW

File: lib/python/rose/macro.py [hunk 1]
```diff
@@ -50,6 +50,8 @@
 ERROR_LOAD_CHOSEN_META_PATH = "Could not find metadata for {0}, using {1}\n"
 ERROR_LOAD_META_PATH = "Could not find metadata for {0}"
 ERROR_LOAD_CONF_META_NODE = "Error: could not find meta flag"
+ERROR_MACRO_CASE_MISMATCH = "Error: Namelist option mismatch; \n {0} does not\
```
Comment (benfitzpatrick) [inline lib/python/rose/macro.py]: Brackets are better for continuation lines

File: lib/python/rose/formats/namelist.py [hunk 2]
```diff
@@ -389,4 +389,4 @@ def validate_config(config, meta_config, add_report_func):
             is_error = (option.lower() != option)
         if is_error:
             add_report_func(section, option, value,
-                            ERROR_UPPERCASE, is_warning=True)
+                            ERROR_UPPERCASE, is_warning=False)
```
Comment (benfitzpatrick) [inline lib/python/rose/formats/namelist.py]: You can just remove the `is_warning` bit

File: lib/python/rose/macro.py [hunk 3]
```diff
@@ -723,12 +725,18 @@ def pretty_format_config(config):
...
             values = rose.variable.array_split(node.value, ",")
-            node.value = pretty_format_value(values)   
+            node.value = pretty_format_value(values)
             new_keylist = pretty_format_keys(keylist)
-            if new_keylist != keylist:
+            if new_keylist != keylist and ignore_error is False:
                 s_node.unset(keylist)
                 s_node.set(new_keylist, node.value, node.state, node.comments)
-                
+                _report_error(text=ERROR_MACRO_CASE_MISMATCH.format(keylist[1],
```
Comment (benfitzpatrick) [inline lib/python/rose/macro.py]: It's simpler if the `if` is around this part of the block and the sys.exit, I think.

File: lib/python/rose/macro.py [hunk 4]
```diff
@@ -50,6 +50,8 @@
 ERROR_LOAD_CHOSEN_META_PATH = "Could not find metadata for {0}, using {1}\n"
 ERROR_LOAD_META_PATH = "Could not find metadata for {0}"
 ERROR_LOAD_CONF_META_NODE = "Error: could not find meta flag"
+ERROR_MACRO_CASE_MISMATCH = "Error: Namelist option mismatch; \n {0} does not\
+ match {1}, please only use lowercase."
```
Comment (benfitzpatrick) [inline lib/python/rose/macro.py]: Lowercase is just for namelists - we might insist that environment variables be uppercase or something in the future.

**Comment by kaday** [comment]
@benfitzpatrick Agree, all done and tested. Back to you.

**Pull request merged**